source/secure_tunneling_operations.c

```diff
@@ -15,9 +15,48 @@
 
 #define INVALID_STREAM_ID 0
 
+/*********************************************************************************************************************
+ * SERVICE AND CONNECTION ID HASH TABLE
+ ********************************************************************************************************************/
+
+static const uint32_t s_bit_scrambling_magic = 0x45d9f3bU;
+static const uint32_t s_bit_shift_magic = 16U;
+
+/* this is a repurposed hash function based on the technique in splitmix64. The magic number was a result of numerical
+ * analysis on maximum bit entropy. */
+uint64_t aws_secure_tunnel_hash_connection_id(const void *to_hash) {
```

Comment:
I found that the same pattern was being used and implemented for a hash table using a uint32_t stream id as the key in event stream and figured I'd defer to experience in how I should handle it here. The id can be any uint32 number, in any order they decide. There is zero expected id sequence. It's probably safe to assume they'd do something reasonable like start at 1 and tick up as needed but who can say if that's what someone will do?

Reply:
Had a discussion and it's safer to hash to reduce possibility of collision. What we should do is probably move this hash function into common since we currently don't have a basic hash function for ints there.